libethcore/Transaction.cpp

```diff
@@ -147,3 +159,29 @@ bigint TransactionBase::gasRequired(bool _contractCreation, bytesConstRef _data,
 	return ret;
 }
 
+h256 TransactionBase::sha3(IncludeSignature _sig) const
+{
+	if (_sig == WithSignature && m_hashWith)
+		return m_hashWith;
+
+	RLPStream s;
+	if (m_chainId > 0 && m_type != NullTransaction) //EIP155 is On
+	{
+		//!!! code copy from streamRLP!  cant change streamRLP because it should not be affected by EIP155 proposal (it affects only hash)
```

Comment:
helper is fine. 
I do not think that hash calculation should be puted in streamRLP
streamRLP means stream the transaction data into RLP. 
streamRLPforEIP155 better be a helper